app/src/main/java/com/seafile/seadroid2/data/DataManager.java

```diff
@@ -929,4 +1023,103 @@ public String search(String query, int page) throws SeafException {
             return null;
         }
     }
+
+    private FileBlocks chunkFile(String encKey, String enkIv, String filePath) throws IOException {
+        File file = new File(filePath);
+        InputStream in = null;
+        OutputStream out = null;
+        byte[] buffer = new byte[BUFFER_SIZE];
+        FileBlocks seafBlock = new FileBlocks();
+        try {
+            in = new FileInputStream(file);
+
+            // Log.d(DEBUG_TAG, "file size " + file.length());
+            while (in.read(buffer, 0, BUFFER_SIZE) != -1) {
+                final byte[] cipher = Crypto.encrypt(buffer, encKey, enkIv);
+                final String blkid = Crypto.sha1(cipher);
+                seafBlock.chunks.add(cipher);
```

Comment:
`seafBlock.chunks.add(cipher);`

this will consume a lot of memory (probably too much on older devices). Also, it seems to be unnecessary, as that array is never accessed again?